File: src/logicServer/src/main/java/via/sep3/logicserver/shared/WorkoutDTO.java
package via.sep3.logicserver.shared;

import java.util.List;

import org.springframework.stereotype.Component;

import lombok.Data;

@Data
@Component
public class WorkoutDTO {
    private int id;
    private String Name;
    private String Description;
    private int DurationInMin;
    private int CreatedBy;
    private int FollowedBy;
    private boolean isPublic;
    private List<ExerciseDTO> Exercises;

    public int numberOfExercises() {
        return Exercises.size();
    }
}

File: src/webServer/Model/DTOs/WorkoutDTO.cs
using Shared.DTOs;

namespace Model.DTOs;

public class WorkoutDTO
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int DurationInMin { get; set; }
    public int NumberOfExercises()
    {
        return Exercises.Count;
    }
    public int CreatedBy { get; set; }
    public int FollowedBy { get; set; }
    public bool IsPublic { get; set; }
    public List<ExerciseDTO>? Exercises { get; set; }
    public List<FollowWorkoutDTO>? Followers;
}